pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentDispatcherMultipleConsumers.java

```diff
@@ -194,7 +193,9 @@ public void sendMessages(List<Entry> entries) {
             TOTAL_AVAILABLE_PERMITS_UPDATER.addAndGet(this, -consumer.sendMessages(entries).getTotalSentMessages());
         } else {
             entries.forEach(entry -> {
-                int totalMsgs = getBatchSizeforEntry(entry.getDataBuffer(), subscription, -1);
+                MessageMetadata msgMetatada = Consumer.peekMessageMetadata(entry.getDataBuffer(), subscription, -1);
```

Comment:
I see this pattern is used in multiple places. It would be good that you can still keep `getBatchSizeforEntry` and make it call `peekMessageMetadata` and return the total message size.

Reply:
Will fix